src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java

```diff
@@ -29,6 +30,19 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
 
         final TFM_PlayerData data = TFM_PlayerData.getPlayerData(sender_p);
         data.setTag(tag.toString());
+        
+        if (!TFM_AdminList.isSuperAdmin(sender))
+        {
+            for (String word : Command_tag.FORBIDDEN_WORDS)
+            {
+                if (args[0].toLowerCase().contains(word))
+                {
+                    playerMsg("That tag contains a forbidden word.");
+                    TFM_PlayerData.getPlayerData(sender_p).setTag(null);
```

Comment:
I would suggest to either you or @JeromSar to create a removeTag option within the player data, just a bit cleaner.